client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java

```diff
@@ -56,6 +68,39 @@
     private NodeSelector nodeSelector = NodeSelector.ANY;
     private boolean strictDeprecationMode = false;
     private boolean compressionEnabled = false;
+    private boolean metadataHeaderEnabled = true;
```

Comment:
metaHeaderEnabled to conform with the `meta header` noun

Reply:
Done. Also renamed constants.